### ISSUE TITLE
malloc error on mac

### ISSUE THREAD

**Issue by MarkehMe** [katmagic/Shallot#27]
I was able to compile Shallot on macOS 10.12 after installing the OpenSSL package [like this article says](https://www.anintegratedworld.com/mac-osx-fatal-error-opensslsha-h-file-not-found/), heres the output: http://pastebin.com/GBM3CnE8

However. I get "_error: can't allocate region_" when using it: 

```
➜  Shallot git:(master) ./shallot ^test
-----------------------------------------------------------------
Found matching domain after 1424617 tries: test6bd7snbmzwch.onion
-----------------------------------------------------------------
shallot(10891,0x700001da9000) malloc: *** mach_vm_map(size=3458764513820545024) failed (error code=3)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
[1]    10891 segmentation fault  ./shallot ^test
```

**Comment by TomCorwine** [comment]
In order for me to get this to build on macOS 10.11 and 10.12, I had to add the following to `CFLAGS` in order to point at the right OpenSSL installation (which was installed using Homebrew):
`-I/usr/local/opt/openssl/include`
`-L/usr/local/opt/openssl/lib`

So line 18 of Makefile would look like this:
`CFLAGS:=$(CFLAGS) -O3 -I/usr/include -I/usr/local/include -I/usr/local/opt/openssl/include -L/usr/lib -L/usr/local/lib -L/usr/local/opt/openssl/lib`

@MarkehMe Any chance this helps?

**Comment by MarkehMe** [comment]
Looks like the solution for now! 

They should have something added to the Makefile for this, or perhaps there is a better way to detect the path of openssl (not sure if this project is even maintained though) 

e.g:
```makefile
OS_DETECTED := $(shell gcc -dumpmachine)
ifneq (, $(findstring darwin, $(OS_DETECTED)))
    CFLAGS:=$(CFLAGS) -I/usr/local/opt/openssl/include -L/usr/local/opt/openssl/lib
endif
```

**Comment by TomCorwine** [comment]
Yeah, I was thinking of applying some of these pull requests to my fork.

**Comment by nashe** [comment]
This issue has been fixed in another fork several months ago, see ReclaimYourPrivacy/eschalot#11 :-)
